src/lib/intent/composed-intent.ts

```diff
@@ -134,6 +137,24 @@ export default class ComposedIntent implements Intent {
     return true
   }
 
+  isTagIntent (): boolean {
+    // TODO Is this a viable option? How can we be sure that composed
+    // intents are not a mix of ct intents and tag intents?
```

Comment:
I think in theory you can only compose intents together that lead to being combined into the same request.  
The concern is still valid in that one could make a mistake in the grouping logic and have those combined together.  
But I think otherwise this should not be a problem.

Reply:
Do you mean the internal grouping logic that relies on the `groupsWith` and the `endsGroup` methods? So with regard to this worry we should not add an extra check somewhere, but just be cautious when we implement those methods?

Reply:
Yes, I would say so as well. The grouping is only there to keep the amount of HTTP requests as small as possible. 

Also: when a group of intents contains at least one tag intent, it should fulfull the `isTagIntent`. If you have mixed intent types in there, the e.g. `isContentType` check would succeed as well. 

So it's up to the grouping logic to be "colocated" with / know about the HTTP adapter for the available endpoints and thus how to group things.